framework/src/modules/chain/submodules/loader.js

```diff
@@ -184,70 +185,54 @@ __private.syncTimer = function() {
 };
 
 /**
- * Gets a random peer and loads signatures from network.
- * Processes each signature from peer.
+ * Loads signatures from network.
+ * Processes each signature from the network.
  *
  * @private
  * @param {function} cb
  * @returns {setImmediateCallback} cb, err
  * @todo Add description for the params
  */
-__private.loadSignatures = function(cb) {
-	async.waterfall(
-		[
-			function(waterCb) {
-				self.getRandomPeerFromNetwork((getRandomPeerFromNetworkErr, peer) => {
-					if (getRandomPeerFromNetworkErr) {
-						return setImmediate(waterCb, getRandomPeerFromNetworkErr);
-					}
-					return setImmediate(waterCb, null, peer);
-				});
-			},
-			function(peer, waterCb) {
-				library.logger.info(`Loading signatures from: ${peer.string}`);
-				peer.rpc.getSignatures((err, res) => {
-					if (err) {
-						modules.peers.remove(peer);
-						return setImmediate(waterCb, err);
-					}
-					return library.schema.validate(
-						res,
-						definitions.WSSignaturesResponse,
-						validateErr => setImmediate(waterCb, validateErr, res.signatures)
-					);
+__private.loadSignatures = async function() {
```

Comment:
We should cover now this with unit tests.

Reply:
I think it's better to focus more on functional tests at this stage because the implementation details are still not in a settled state.

Reply:
If the implementation details should be clear enough because what we want it's to replicate the exact behaviour that it's currently working.
Anyways, I might agree but then I miss new functional test cases in the PR and we cannot skip any network tests.